include/tvm/runtime/relax_vm/memory_manager.h

```diff
@@ -71,6 +72,13 @@ class Allocator {
    *  \return A sized allocation in the form of a buffer.
    */
   virtual Buffer Alloc(size_t nbytes, size_t alignment, DLDataType type_hint) = 0;
+  /*! \brief Allocate a buffer given a size, alignment and type.
+   * \param shape The shape of allocated tensor.
+   *  \param type_hint A type hint to the allocator.
+   * \param mem_scope The memory scope of allocated tensor.
+   *  \return A sized allocation in the form of a buffer.
+   */
+  virtual Buffer Alloc(ShapeTuple shape, DLDataType type_hint, String mem_scope);
```

Comment:
AllocDataSpaceWithScope to be consistent with the original one

Reply:
In this case type_hint should just be dtype

Reply:
right makes sense, I'll fix that, thanks.

Reply:
Done thanks